services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java

```diff
@@ -854,6 +855,9 @@ public void validateUser(UserModel authenticatedUser) {
         if (authenticatedUser == null) return;
         if (!authenticatedUser.isEnabled()) throw new AuthenticationFlowException(AuthenticationFlowError.USER_DISABLED);
         if (realm.isBruteForceProtected()) {
+            if (event.getEvent().getType().equals(EventType.RESET_PASSWORD)) {
```

Comment:
It's a hack. The events are just there to send events. It's not a generic way to find out what is going on in the authentication process. Not sure exactly how it should be done though. Maybe @patriot1burke has a suggestion.